test/images/image-util.sh

```diff
@@ -117,23 +141,46 @@ push() {
   docker_version_check
   TAG=$(<"${IMAGE}"/VERSION)
   if [[ -f ${IMAGE}/BASEIMAGE ]]; then
-    archs=$(listArchs)
+    base_images=$(listBaseImages)
+    # NOTE(claudiub): if the REMOTE_DOCKER_URL var is not set, or it is an empty string, we must skip
+    # pushing the Windows image and including it into the manifest list.
+    if [[ ! -v "REMOTE_DOCKER_URL" || -z "${REMOTE_DOCKER_URL}" ]]; then
+      if [[ ! -z "$(printf "%s\n" $base_images | grep '^windows')" ]]; then
+        echo "Skipping pushing the image '${IMAGE}' for Windows. REMOTE_DOCKER_URL should be set, containing the URL to a Windows docker daemon."
+        base_images=$(printf "%s\n" $base_images | grep -v "^windows")
```

Comment:
This line looks awfully similar to the conditional inside line 148. Is there a way to somehow make it more DRY?

Reply:
It looks similar, but it is not. The conditional greps ``^windows`` to check if we have to build the image for Windows, while here it is ``grep -v``, which excludes all lines with ``^windows``.